File: 0028_Implement strStr().java
class Solution {
    public int strStr(String haystack, String needle) {
        if (needle.equals("")) return 0;

        int len1 = haystack.length();
        int len2 = needle.length();

        for (int i = 0; i < len1 - len2; i++) {
            for (int j = 0; j < len2; j++) {
                if (haystack.charAt(i + j) != needle.charAt(j)) break;
                if (j == len2 - 1) return i;
            }
        }

        return -1;
    }
}

File: 0028_Implement strStr().go
package main

func strStr(haystack string, needle string) int {
	if len(needle) == 0 {
		return 0
	}

	A, B := len(haystack), len(needle)
	for i := 0; i < A-B+1; i++ {
		j := 0
		for ; j < B; j++ {
			if haystack[i+j] != needle[j] {
				break
			}
		}

		if j == B {
			return i
		}
	}

	return -1
}
